### PR TITLE
VCB-145 Allowed Procedure instructions.

### PR DESCRIPTION
**Pull request by PhilSchni**
solves https://virtualctboard.atlassian.net/browse/VCB-184 as well

### CODE REVIEW

File: src/assembler/parser/index.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ export function parse(code: string): ICodeFile {
       name: 'Whitespace',
       pattern: /\s+/
     },
+    {
+      name: 'ProcedureInstructions',
+      pattern: `PROC\|ENDP\|END`
```
Comment (m-dot-wigger) [inline src/assembler/parser/index.ts:32]: Split PROC and both END and ENDP into 2 rules since label is allowed before PROC but not the END ones.

**Review by m-dot-wigger — CHANGES_REQUESTED** [review]
Nice without matching the line to an instruction no empty instruction encoder is required

File: src/assembler/parser/index.ts (hunk 1 above)
Comment (PhilSchni) [reply src/assembler/parser/index.ts]: Have changed this :).

**Review by m-dot-wigger — APPROVED** [review]
all good now

**Pull request merged**